retrieval/target_test.go

```diff
@@ -390,6 +390,47 @@ func BenchmarkScrape(b *testing.B) {
 	}
 }
 
+func TestURLParams(t *testing.T) {
+	type test struct {
+		metric       string
+		resultNormal clientmodel.Metric
+		resultHonor  clientmodel.Metric
+	}
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", `text/plain; version=0.0.4`)
+				w.Write([]byte{})
+				if r.FormValue("foo") != "bar" {
+					t.Fatalf("URL parameter 'foo' had unexpected value '%v'", r.FormValue("foo"))
+				}
+			},
+		),
+	)
+	defer server.Close()
+	serverURL, err := url.Parse(server.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	target := NewTarget(&config.ScrapeConfig{
+		JobName:        "test_job1",
+		ScrapeInterval: config.Duration(1 * time.Minute),
+		ScrapeTimeout:  config.Duration(1 * time.Second),
+		Scheme:         serverURL.Scheme,
+	},
+		clientmodel.LabelSet{
```

Comment:
The indentation looks kind of exotic here, although I realize gofmt keeps it that way. How about:

```
  target := NewTarget(
    &config.ScrapeConfig{
      JobName:        "test_job1",
      ScrapeInterval: config.Duration(1 * time.Minute),
      ScrapeTimeout:  config.Duration(1 * time.Second),
      Scheme:         serverURL.Scheme,
    },
    clientmodel.LabelSet{
      clientmodel.AddressLabel: clientmodel.LabelValue(serverURL.Host),
      "__param_foo":            "bar",
    },
    nil,
  )
```